app/Http/Client/Alchemy/AlchemyPendingRequest.php

```diff
@@ -217,6 +218,40 @@ public function getWalletNfts(Wallet $wallet, Network $network, string $cursor =
         );
     }
 
+    /**
+     * @see https://docs.alchemy.com/reference/getnftmetadatabatch
+     *
+     * @param  Collection<int, Nft>  $nfts
+     */
+    public function nftMetadata(Collection $nfts, CollectionModel $collection): Web3NftsChunk
```

Comment:
As we discussed in the Slack: 
`CollectionModel $collection`  needs to be replaced by `Network $network`